examples/feddg_ga_example/server.py

```diff
@@ -0,0 +1,99 @@
+import argparse
+from functools import partial
+from typing import Any, Dict, Optional
+
+import flwr as fl
+from flwr.common.typing import Config
+
+from examples.models.cnn_model import MnistNetWithBnAndFrozen
+from examples.utils.functions import make_dict_with_epochs_or_steps
+from fl4health.client_managers.fixed_sampling_client_manager import FixedSamplingClientManager
+from fl4health.model_bases.apfl_base import ApflModule
+from fl4health.server.base_server import FlServer
+from fl4health.strategies.feddg_ga_strategy import FedDgGaStrategy
+from fl4health.utils.config import load_config
+from fl4health.utils.functions import get_all_model_parameters
+from fl4health.utils.metric_aggregation import evaluate_metrics_aggregation_fn, fit_metrics_aggregation_fn
+from fl4health.utils.random import set_all_random_seeds
+
+
+def fit_config(
+    batch_size: int,
+    n_server_rounds: int,
+    current_round: int,
+    local_epochs: Optional[int] = None,
+    local_steps: Optional[int] = None,
+    evaluate_after_fit: bool = False,
+) -> Config:
+    return {
+        **make_dict_with_epochs_or_steps(local_epochs, local_steps),
+        "current_server_round": current_round,
+        "batch_size": batch_size,
+        "n_server_rounds": n_server_rounds,
+        "evaluate_after_fit": evaluate_after_fit,
+    }
+
+
+def main(config: Dict[str, Any]) -> None:
+    # This function will be used to produce a config that is sent to each client to initialize their own environment
+    fit_config_fn = partial(
+        fit_config,
+        config["batch_size"],
+        config["n_server_rounds"],
+        local_epochs=config.get("local_epochs"),
+        local_steps=config.get("local_steps"),
+        evaluate_after_fit=config.get("evaluate_after_fit", False),
+    )
+
+    initial_model = ApflModule(MnistNetWithBnAndFrozen())
+
+    # Server performs simple FedAveraging as its server-side optimization strategy
+    strategy = FedDgGaStrategy(
+        min_fit_clients=config["n_clients"],
+        min_evaluate_clients=config["n_clients"],
+        # Server waits for min_available_clients before starting FL rounds
+        min_available_clients=config["n_clients"],
+        on_fit_config_fn=fit_config_fn,
+        # We use the same fit config function, as nothing changes for eval
+        on_evaluate_config_fn=fit_config_fn,
+        fit_metrics_aggregation_fn=fit_metrics_aggregation_fn,
+        evaluate_metrics_aggregation_fn=evaluate_metrics_aggregation_fn,
+        initial_parameters=get_all_model_parameters(initial_model),
+    )
+
+    client_manager = FixedSamplingClientManager()
+    server = FlServer(strategy=strategy, client_manager=client_manager)
```

Comment:
One thing we might want to think about here is either:
1. We put a check in the FlServer class to ensure that when the strategy being used is FedDgGaStrategy, then the client manager must have type `FixedSamplingClientManager`.
2. We create a server class on top of FlServer that enforces that explicitly.

That way someone doesn't accidentally use a different client manager and therefore get weird but hard to debug results?

Reply:
Ah I see, you actually do this assertion in the strategy. That's even better. Feel free to ignore.